files/__addonLocation__/package.json

```diff
@@ -10,18 +10,25 @@
   "author": "",
   "files": [
     "addon-main.cjs",
-    "dist"
+    "dist",
+    "declarations"
   ],
-  "scripts": {
-    "build": "rollup --config",
+  "scripts": {<% if (typescript) { %>
+    "build": "concurrently 'npm:build:*'",
+    "build:js": "rollup --config",
+    "build:types": "glint --build", <% } else { %>
+    "build": "rollup --config", <% } %>
     "lint": "concurrently 'npm:lint:*(!fix)' --names 'lint:'",
     "lint:fix": "concurrently 'npm:lint:*:fix' --names 'fix:'",
     "lint:hbs": "ember-template-lint . --no-error-on-unmatched-pattern",
     "lint:js": "eslint . --cache",
     "lint:hbs:fix": "ember-template-lint . --fix --no-error-on-unmatched-pattern",
     "lint:js:fix": "eslint . --fix",<% if (typescript) { %>
-    "lint:types": "glint",<% } %>
-    "start": "rollup --config --watch",
+    "lint:types": "glint",
```

Comment:
Why does this sometimes use `--build` and other times not?

Reply:
Good callout, I missed that!

I tested this in my addon, and came up with the changes in [this commit](https://github.com/simonihmig/ember-stargate/pull/655/commits/31c3437a494b3d11efd9a3bc03a3144fe87d4b83):
* remove `--build`
* remove all compilerOptions except for `declarationDir`. The other don't seem to be needed AFAICT, maybe implied by `-d`?
* added `-d` to the `build:types` script. Having `"declaration": true` alone does not seem to work, `-d` is strictly needed for glint to output declaration files, is that right?

Reply:
> The other don't seem to be needed AFAICT, maybe implied by -d?

They may be set to reasonable defaults in `@tsconfig/ember`

> Having "declaration": true alone does not seem to work, -d is strictly needed for glint to output declaration files, is that right?

It's arguably a bug in Glint that we _require_ you to run with `--declaration` to emit `.d.ts` files. I think the reason we landed there is that there's some hazy gray area in terms of what it means for us to match `tsc`'s own behavior, since we effectively hard-code `noEmit: true`, which implies certain things about how other config like `declarations` behaves.

Regardless, you're correct that you have to use the `--declaration`/`-d` flag for the Glint CLI to emit `.d.ts` files.

Looking at that commit you linked, I think you also probably want `-d` in your `start:types` script so that that will update your `.d.ts` files as your sources change. Otherwise looks reasonable to me! 👍